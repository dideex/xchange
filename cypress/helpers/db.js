const mongoose = require('mongoose')
const mongoConfig = require('./config.json').mongodb

module.exports = mongoose.connect(mongoConfig, {useNewUrlParser: true})
