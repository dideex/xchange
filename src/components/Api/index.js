import {noty} from '../common'

export class Api {
  constructor() {
    this.backend = 'http://localhost:3030'
    this.isNetworkError = false
  }

  post(endpoint, data = {}, token = null) {
    return fetch(`${this.backend}/api/${endpoint}`, {
      method: 'POST',
      headers: {
        Accept: 'application/json',
        'Content-Type': 'application/json',
        Authorization: `bearer ${token}`,
      },
      body: JSON.stringify(data),
    }).then(response => response.json())
  }

  get(endpoint, params = '', token = null) {
    return fetch(`${this.backend}/api/${endpoint}${params}`, {
      method: 'GET',
      headers: {
        Accept: 'application/json',
        'Content-Type': 'application/json',
        Authorization: `bearer ${token}`,
      },
    }).then(response => response.json())
  }
  
  errorEmitter = fn => ({err, errCode, ...data}) => {
    if (err) {
      this.isNetworkError = true
      console.error('errorEmitter ___ ', errCode)
      noty(err, 'error')
    } else fn(data)
  }
}

export default new Api()
