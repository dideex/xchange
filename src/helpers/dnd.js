import React from 'react'
import {DragDropContext} from 'react-dnd'
import TestBackend from 'react-dnd-test-backend'

export function wrapInTestContext(DecoratedComponent) {
  const TestStub = props => <DecoratedComponent {...props} />
  return DragDropContext(TestBackend)(TestStub)
}
