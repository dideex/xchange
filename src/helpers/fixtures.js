export const fakeCurrnecy = [
  {
    order: 1,
    _id: 1,
    icon: 'eth',
    id: 'eth',
    name: 'tether',
    label: 'tether label',
    source: 'tether source wallet'
  },
  {
    order: 2,
    _id: 2,
    icon: 'bitcoin',
    id: 'bitcoin',
    name: 'lisk',
    label: 'lisk label',
    source: 'lisk source wallet'
  },
  {
    order: 3,
    _id: 3,
    icon: 'rur',
    id: 'rur',
    name: 'litecoin',
    label: 'litecoin label',
    source: 'litecoin source wallet'
  },
]


export const fakeData = {
  id: 'Test id',
  inputValue: '1000',
  toWallet: 'Test wallet',
  currencyInputLabel: 'Test input label',
  currencyOutputLabel: 'Test output label',
  fromWallet: 'Tests fromWallet',
  outputValue: '2000',
  email: 'Tests email',
  username: 'Tests username',
  paymentStatus: 1,
  loading: false,
  wallets: {
    eth: 'eth wallet',
    bitcoint: 'bitcoint wallet',
    rur: 'rur wallet',
  },
  login: 'Test login',
  updatePaymentStatus() {},
}

export const fakeUser = {
  username: 'Fake username',
  password: 'Fake password',
  email: 'fake@email.com',
  phone: '89145678901',
  fio: 'Fake fio',
  token: 'Fake token',
}

export const fakeBitcoinAddress = '3JQSigWTCHyBLRD979JWgEtWP5YiiFwcQB '
export const fakeCreditAddress = '1234 4321 5678 8765'

export const fakeWallets = {
  eth: 'Fake eth wallet',
  bitcoin: 'Fake bitcoin wallet',
  rur: 'Fake rur wallet'
}