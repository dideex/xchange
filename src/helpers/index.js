export {fakeCurrnecy} from './fixtures'

export const delay = (fn = () => {}, timeout = 0) =>
  new Promise(res =>
    setTimeout(() => {
      fn()
      res()
    }, timeout),
  )
