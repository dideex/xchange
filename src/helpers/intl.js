import React from 'react'
import {mount, shallow} from 'enzyme'
import {IntlProvider, intlShape} from 'react-intl'

import messages from '../locale/en.json'
const intlProvider = new IntlProvider({locale: 'en', messages}, {})
const {intl} = intlProvider.getChildContext()

function nodeWithIntlProp(node) {
  return React.cloneElement(node, {intl})
}

export function mountWithIntl(
  node,
  {context, childContextTypes, ...additionalOptions} = {},
) {
  return mount(nodeWithIntlProp(node), {
    context: Object.assign({}, context, {intl}),
    childContextTypes: Object.assign({}, {intl: intlShape}, childContextTypes),
    ...additionalOptions,
  })
}
