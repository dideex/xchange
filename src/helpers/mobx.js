import React from 'react'
import {Provider} from 'mobx-react'
import {lastOperations, cash, user} from '../store'

jest.mock('../components/Api', () => ({
  post: () => Promise.resolve({data: {}}),
  get: () => Promise.resolve({data: {}}),
  errorEmitter: jest.fn(data => fn => data(fn)),
}))

// jest.mock('../components/common', () => ({
//   noty: () => {},
//   Loading: () => <div>Loading</div>,
// }))

export const MobxProvider = ({children, ...stores}) => {
  return (
    <Provider
      lastOperationsStore={lastOperations}
      cashStore={cash}
      userStore={user}
      {...stores}
    >
      {children}
    </Provider>
  )
}
