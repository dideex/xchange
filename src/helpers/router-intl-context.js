import {mount} from 'enzyme'
import {BrowserRouter} from 'react-router-dom'
import {IntlProvider, intlShape} from 'react-intl'
import {shape} from 'prop-types'

import messages from '../locale/en.json'

export const testId = 'test id'

// Instantiate router context
const router = {
  history: {...new BrowserRouter().history},
  route: {
    location: {
      path: '/location',
      href: '/location',
    },
    match: {
      path: '/location',
      params: {
        id: testId,
      },
    },
  },
}

// const createContext = () => ({
//   context: {router},
//   childContextTypes: {router: shape({})},
// })

const intlProvider = new IntlProvider({locale: 'en', messages}, {})
const {intl} = intlProvider.getChildContext()

const createMixedContext = ({
  router,
  context,
  childContextTypes,
  ...additionalOptions
} = {}) => ({
  context: Object.assign({}, {router}, context, {intl}),
  childContextTypes: Object.assign(
    {},
    {router: shape({})},
    {intl: intlShape},
    childContextTypes,
  ),
  ...additionalOptions,
})

export function mountWrap(node) {
  return mount(node, createMixedContext({router}))
}

// export function shallowWrap(node) {
//   return shallow(node, createContext())
// }
