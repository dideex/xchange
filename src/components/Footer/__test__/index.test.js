import React from 'react'
import Component from '../index'
import {shallow} from 'enzyme'
import {mountWithIntl} from '../../../helpers/intl'
import {fakeUser} from '../../../helpers/fixtures'
import Api from '../../Api'
import {delay} from '../../../helpers'

jest.mock('../../../components/Api', () => ({
  post: () => Promise.resolve({data: {}}),
  get: () => Promise.resolve({data: {}, eth: {}}),
  errorEmitter: fn => data => fn(data),
}))

describe('Footer tests', () => {
  describe('Markup', () => {
    const wrapper = mountWithIntl(<Component />)
    it('Basic markup', () => {
      expect(wrapper.html()).toMatchSnapshot()
    })
    it('Loading markup', () => {
      wrapper.find('Footer').setState({loading: true})
      expect(wrapper.html()).toMatchSnapshot()
    })
    it('Email should change', () => {
      wrapper
        .find('input[placeholder="Email"]')
        .simulate('change', {target: {value: fakeUser.email}})
      expect(wrapper.find('InputComponent[placeholder="Email"]').html()).toMatchSnapshot()
    })
    it('Email should show an error after enter wrong email', () => {
      wrapper
        .find('input[placeholder="Email"]')
        .simulate('change', {target: {value: fakeUser.email + '@'}})
      expect(wrapper.find('InputComponent[placeholder="Email"]').html()).toMatchSnapshot()
    })
    it('Phone should change', () => {
      wrapper
        .find('input[placeholder="Phone"]')
        .simulate('change', {target: {value: fakeUser.phone}})
      expect(wrapper.find('InputComponent[placeholder="Phone"]').html()).toMatchSnapshot()
    })
    it('Phone should show an error after enter wrong email', () => {
      wrapper
        .find('input[placeholder="Phone"]')
        .simulate('change', {target: {value: '123'}})
      expect(wrapper.find('InputComponent[placeholder="Phone"]').html()).toMatchSnapshot()
    })
    it('Textarea should change', () => {
      wrapper.find('textarea').simulate('change', {target: {value: fakeUser.username}})
      expect(wrapper.find('textarea').html()).toMatchSnapshot()
    })
  })
  describe('Component behavriour', () => {
    it('Should submit if everything was ok', async () => {
      const wrapper = mountWithIntl(<Component />)
      Api.post = jest.fn(() => Promise.resolve())
      wrapper
        .find('input[placeholder="Email"]')
        .simulate('change', {target: {value: fakeUser.email}})
      wrapper
        .find('input[placeholder="Phone"]')
        .simulate('change', {target: {value: fakeUser.phone}})
      wrapper.find('textarea').simulate('change', {target: {value: fakeUser.username}})
      wrapper.find('button').simulate('click')

      await delay()
      expect(Api.post).toHaveBeenCalledTimes(1)
      expect(Api.post).toHaveBeenCalledWith('sendMessage', {
        email: fakeUser.email,
        message: fakeUser.username,
        phone: "+8(914)-567-89-01"
      })
    })
    it('Should not submit if something was wrong', async () => {
      const wrapper = mountWithIntl(<Component />)
      Api.post = jest.fn(() => Promise.resolve())
      wrapper
        .find('input[placeholder="Email"]')
        .simulate('change', {target: {value: fakeUser.email}})
      wrapper.find('textarea').simulate('change', {target: {value: fakeUser.username}})
      wrapper.find('button').simulate('click')

      await delay()
      expect(Api.post).toHaveBeenCalledTimes(0)
    })
  })
})
