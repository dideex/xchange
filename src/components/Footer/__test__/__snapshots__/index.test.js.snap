// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer tests Markup Basic markup 1`] = `"<footer class=\\"css-n0qkhb\\"><section class=\\"css-17weg8a\\"><h2 class=\\"css-x184e\\"><span>Any questions?</span></h2><label class=\\"css-gudk1o\\"><input placeholder=\\"Email\\" pattern=\\"^(([^<>()\\\\[\\\\]\\\\.,;:\\\\s@&quot;]+(\\\\.[^<>()\\\\[\\\\]\\\\.,;:\\\\s@&quot;]+)*)|(&quot;.+&quot;))@((\\\\[[0-9]{1,3}\\\\.[0-9]{1,3}\\\\.[0-9]{1,3}\\\\.[0-9]{1,3}\\\\])|(([a-zA-Z\\\\-0-9]+\\\\.)+[a-zA-Z]{2,}))$\\" type=\\"text\\" mask=\\"\\" class=\\"css-18b8ap6\\" value=\\"\\"></label><label class=\\"css-gudk1o\\"><input placeholder=\\"Phone\\" pattern=\\"^\\\\d+$\\" mask=\\"phone\\" type=\\"text\\" class=\\"css-18b8ap6\\" value=\\"\\"></label><textarea placeholder=\\"Your message\\" class=\\"css-14ecz31\\"></textarea><button cursor=\\"pointer\\" class=\\"css-8hkfp7\\"><div class=\\"css-1v1d0dg\\">Send</div></button></section><div class=\\"css-13his2v\\"><svg style=\\"width: 100%;\\" class=\\"\\" viewBox=\\"0 0 1440 295\\" fill=\\"none\\" xmlns=\\"http://www.w3.org/2000/svg\\"><path style=\\"fill: #40c0cb;\\" d=\\"M1439.5 294.5H0.5C263.303 123.383 457.877 36.3363 1439.5 0.5V294.5Z\\"></path></svg><div class=\\"css-1u1xm4y\\"><strong>XChange © 2009 - 2018</strong><a href=\\"https://vk.com/didex\\" rel=\\"noopener noreferrer\\" target=\\"_blank\\">Created by DiDex</a></div></div></footer>"`;

exports[`Footer tests Markup Email should change 1`] = `"<label class=\\"css-gudk1o\\"><input placeholder=\\"Email\\" pattern=\\"^(([^<>()\\\\[\\\\]\\\\.,;:\\\\s@&quot;]+(\\\\.[^<>()\\\\[\\\\]\\\\.,;:\\\\s@&quot;]+)*)|(&quot;.+&quot;))@((\\\\[[0-9]{1,3}\\\\.[0-9]{1,3}\\\\.[0-9]{1,3}\\\\.[0-9]{1,3}\\\\])|(([a-zA-Z\\\\-0-9]+\\\\.)+[a-zA-Z]{2,}))$\\" type=\\"text\\" mask=\\"\\" class=\\"css-18b8ap6\\" value=\\"fake@email.com\\"></label>"`;

exports[`Footer tests Markup Email should show an error after enter wrong email 1`] = `"<label class=\\"css-gudk1o\\"><input placeholder=\\"Email\\" pattern=\\"^(([^<>()\\\\[\\\\]\\\\.,;:\\\\s@&quot;]+(\\\\.[^<>()\\\\[\\\\]\\\\.,;:\\\\s@&quot;]+)*)|(&quot;.+&quot;))@((\\\\[[0-9]{1,3}\\\\.[0-9]{1,3}\\\\.[0-9]{1,3}\\\\.[0-9]{1,3}\\\\])|(([a-zA-Z\\\\-0-9]+\\\\.)+[a-zA-Z]{2,}))$\\" type=\\"text\\" mask=\\"\\" class=\\"css-ykrnj4\\" value=\\"fake@email.com@\\"><span class=\\"css-1j8ju8e error--enter error--enter-active\\">Enter your Email</span></label>"`;

exports[`Footer tests Markup Loading markup 1`] = `"<footer class=\\"css-n0qkhb\\"><section class=\\"css-17weg8a\\"><h2 class=\\"css-x184e\\"><span>Any questions?</span></h2><label class=\\"css-gudk1o\\"><input placeholder=\\"Email\\" pattern=\\"^(([^<>()\\\\[\\\\]\\\\.,;:\\\\s@&quot;]+(\\\\.[^<>()\\\\[\\\\]\\\\.,;:\\\\s@&quot;]+)*)|(&quot;.+&quot;))@((\\\\[[0-9]{1,3}\\\\.[0-9]{1,3}\\\\.[0-9]{1,3}\\\\.[0-9]{1,3}\\\\])|(([a-zA-Z\\\\-0-9]+\\\\.)+[a-zA-Z]{2,}))$\\" type=\\"text\\" mask=\\"\\" class=\\"css-18b8ap6\\" value=\\"\\"></label><label class=\\"css-gudk1o\\"><input placeholder=\\"Phone\\" pattern=\\"^\\\\d+$\\" mask=\\"phone\\" type=\\"text\\" class=\\"css-18b8ap6\\" value=\\"\\"></label><textarea placeholder=\\"Your message\\" class=\\"css-14ecz31\\"></textarea><button cursor=\\"pointer\\" class=\\"css-8hkfp7\\"><div class=\\"css-1v1d0dg\\"><div width=\\"32px\\" height=\\"50\\" class=\\"css-1nubzf5\\"><svg class=\\"loading\\" width=\\"32\\" height=\\"32\\" viewBox=\\"0 0 256 256\\" fill=\\"none\\" xmlns=\\"http://www.w3.org/2000/svg\\"><path class=\\"clockwise-big\\" d=\\"M223.416 97.9999C218.569 82.6432 210.093 68.6803 198.707 57.2934C187.32 45.9065 173.357 37.4308 158 32.5839V3.56995C180.899 9.10903 201.828 20.8534 218.487 37.5126C235.147 54.1717 246.891 75.1008 252.43 97.9999H223.416ZM98 32.5839C82.6431 37.4305 68.68 45.9061 57.2931 57.293C45.9062 68.68 37.4306 82.643 32.584 97.9999H3.57001C9.10888 75.1007 20.8532 54.1715 37.5124 37.5123C54.1715 20.8531 75.1007 9.10882 98 3.56995V32.5839ZM32.584 158C37.4309 173.357 45.9066 187.32 57.2935 198.706C68.6804 210.093 82.6432 218.569 98 223.416V252.43C75.1008 246.891 54.1717 235.146 37.5126 218.487C20.8535 201.828 9.10909 180.899 3.57001 158H32.584ZM158 223.416C173.357 218.569 187.319 210.093 198.706 198.706C210.093 187.319 218.569 173.357 223.416 158H252.43C246.89 180.899 235.146 201.828 218.487 218.487C201.828 235.146 180.899 246.89 158 252.43V223.416Z\\" style=\\"fill: #40c0cb;\\"></path><path class=\\"counter-clockwise\\" d=\\"M193.942 107.266C190.592 96.6555 184.735 87.0081 176.868 79.1403C169 71.2725 159.353 65.4159 148.742 62.066V42C164.568 45.8289 179.032 53.9465 190.545 65.4606C202.057 76.9747 210.173 91.4397 214 107.266H193.942ZM107.266 62.066C96.6555 65.4159 87.0081 71.2725 79.1403 79.1403C71.2725 87.0081 65.4159 96.6555 62.066 107.266H42C45.8276 91.4387 53.9446 76.973 65.4588 65.4588C76.973 53.9446 91.4387 45.8276 107.266 42V62.058V62.066ZM62.066 148.742C65.4159 159.353 71.2725 169 79.1403 176.868C87.0081 184.735 96.6555 190.592 107.266 193.942V214C91.4387 210.172 76.973 202.055 65.4588 190.541C53.9446 179.027 45.8276 164.561 42 148.734H62.058L62.066 148.742ZM148.742 193.942C159.353 190.592 169 184.735 176.868 176.868C184.735 169 190.592 159.353 193.942 148.742H214C210.171 164.568 202.054 179.032 190.539 190.545C179.025 202.057 164.56 210.173 148.734 214V193.942H148.742Z\\" style=\\"fill: #40c0cb;\\"></path></svg></div>Send</div></button></section><div class=\\"css-13his2v\\"><svg style=\\"width: 100%;\\" class=\\"\\" viewBox=\\"0 0 1440 295\\" fill=\\"none\\" xmlns=\\"http://www.w3.org/2000/svg\\"><path style=\\"fill: #40c0cb;\\" d=\\"M1439.5 294.5H0.5C263.303 123.383 457.877 36.3363 1439.5 0.5V294.5Z\\"></path></svg><div class=\\"css-1u1xm4y\\"><strong>XChange © 2009 - 2018</strong><a href=\\"https://vk.com/didex\\" rel=\\"noopener noreferrer\\" target=\\"_blank\\">Created by DiDex</a></div></div></footer>"`;

exports[`Footer tests Markup Phone should change 1`] = `"<label class=\\"css-gudk1o\\"><input placeholder=\\"Phone\\" pattern=\\"^\\\\d+$\\" mask=\\"phone\\" type=\\"text\\" class=\\"css-18b8ap6\\" value=\\"+8(914)-567-89-01\\"></label>"`;

exports[`Footer tests Markup Phone should show an error after enter wrong email 1`] = `"<label class=\\"css-gudk1o\\"><input placeholder=\\"Phone\\" pattern=\\"^\\\\d+$\\" mask=\\"phone\\" type=\\"text\\" class=\\"css-ykrnj4\\" value=\\"+1(23\\"><span class=\\"css-1j8ju8e error--enter error--enter-active\\">Введите ваш телефон</span></label>"`;

exports[`Footer tests Markup Textarea should change 1`] = `"<textarea placeholder=\\"Your message\\" class=\\"css-14ecz31\\">Fake username</textarea>"`;
